### ISSUE TITLE
Keyboard machine controlls stay active even if "Machine Control" Tab lost focus

### ISSUE THREAD

**Issue by JohnKapri** [winder/Universal-G-Code-Sender#124]
A friend of mine complained about the keyboard controls beeing active, even if the "Machine Control" tab isn't the selected anymore, resulting in movements of the CNC if you try to navigate through your Code in another tab. This once even broke his machine's tool by accident.

Possible fix is to add a check whether controlContextTabbedPane has machineControlPanel as the selected Component in line 1490 in MainWindow.java. That's how I fixed it for him.

Change after line 1490:

```
if (((arrowMovementEnabled.isSelected()) &&
        e.getID() == KeyEvent.KEY_PRESSED) &&
        xPlusButton.isEnabled() &&
        controlContextTabbedPane.getSelectedComponent() ==
        machineControlPanel) {
                // Do stuff
}
```

**Comment by gregoryperrin** [comment]
Just manually typing gcode in a different tab than "Machine Control" (Commands or Macro, for example) makes the machine move when "enable keyboard movement" is checked.

**Comment by jahnj0584** [comment]
i think this issue is gone.

**Comment by breiler** [comment]
Fixed in #2163, the listener will now filter out key events when a JDialog or JTextField has focus.
